Tools/LambdaTestTool-v2/README.md

```diff
@@ -0,0 +1,288 @@
+| `--api-gateway-emulator-mode` | API Gateway mode (Rest/HttpV1/HttpV2) | API Gateway Mode |
+| `--no-launch-window` | Disable auto-launching web interface | Optional |
+
+
+## API Gateway Configuration
+When using API Gateway mode, you need to configure the route mapping using the APIGATEWAY_EMULATOR_ROUTE_CONFIG environment variable. This can be a single route or an array of routes:
+
+### Single Route Configuration
+```
+{
+    "LambdaResourceName": "AddLambdaFunction",
+    "HttpMethod": "Get",
+    "Path": "/add/{x}/{y}"
+}
+```
+
+### Multiple Routes Configuration
+
+```
+[
+    {
+        "LambdaResourceName": "AddLambdaFunction",
+        "HttpMethod": "Get",
+        "Path": "/add/{x}/{y}"
+    },
+    {
+        "LambdaResourceName": "SubtractLambdaFunction",
+        "HttpMethod": "Get",
+        "Path": "/minus/{x}/{y}"
+    }
+]
+
+```
+
+## Example Lambda Function Setup
+
+Here's a simple Lambda function that adds two numbers together.
+
+### 1. Lambda Function Code
+This can be implemented in two ways:
+
+#### Option 1: Using Top-Level Statements
+
+
+```csharp
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.Core;
+using Amazon.Lambda.RuntimeSupport;
+using Amazon.Lambda.Serialization.SystemTextJson;
+
+var Add = (APIGatewayHttpApiV2ProxyRequest request, ILambdaContext context) =>
+{
+    // Parse x and y from the path parameters
+    var x = int.Parse(request.PathParameters["x"]);
+    var y = int.Parse(request.PathParameters["y"]);
+    return (x + y).ToString();
+};
+
+await LambdaBootstrapBuilder.Create(Add, new CamelCaseLambdaJsonSerializer())
+    .Build()
+    .RunAsync();
+
+```
+
+Configure the Lambda function to use the test tool:
+
+**Properties/launchSettings.json**
+```
+{
+  "profiles": {
+    "AspireTestFunction": {
+      "commandName": "Project",
+      "environmentVariables": {
+        "AWS_LAMBDA_RUNTIME_API": "localhost:5050/AddLambdaFunction"
+      }
+    }
+  }
+}
+```
+
+#### Option 2: Using Class Library
+```
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.Core;
+
+namespace MyLambdaFunction;
+
+public class Function
+{
+    public int Add(APIGatewayHttpApiV2ProxyRequest request, ILambdaContext context)
+    {
+        var x = int.Parse(request.PathParameters["x"]);
+        var y = int.Parse(request.PathParameters["y"]);
+        return x + y;
+    }
+}
+```
+
+Configure the Lambda function to use the test tool:
+
+**Properties/launchSettings.json**
+```
+{
+  "profiles": {
+    "LambdaRuntimeClient_FunctionHandler": {
+      "workingDirectory": ".\\bin\\$(Configuration)\\net8.0",
+      "commandName": "Executable",
+      "commandLineArgs": "exec --depsfile ./MyLambdaFunction.deps.json  --runtimeconfig ./MyLambdaFunction.runtimeconfig.json %USERPROFILE%/.dotnet/tools/.store/amazon.lambda.testtool/0.0.2-preview/amazon.lambda.testtool/0.0.2-preview/content/Amazon.Lambda.RuntimeSupport/net8.0/Amazon.Lambda.RuntimeSupport.dll MyLambdaFunction::MyLambdaFunction.Function::Add",
```

Comment:
```
%USERPROFILE%/.dotnet/tools/.store/amazon.lambda.testtool/0.0.2-preview/amazon.lambda.testtool/0.0.2-preview/content/Amazon.Lambda.RuntimeSupport/net8.0/Amazon.Lambda.RuntimeSupport.dll MyLambdaFunction::MyLambdaFunction.Function::Add"
```

wasnt sure if there was an easy way to dynamically set the test tool version based on the current release number? or if there was a more simpler way to write this

Reply:
I think that's the issue that norm was referring to the other day.